packages/aws-cdk-lib/aws-rds/test/cluster.test.ts

```diff
@@ -1798,15 +1819,139 @@ describe('cluster', () => {
         new DatabaseCluster(stack, 'Database', {
           engine: DatabaseClusterEngine.AURORA,
           vpc,
-          writer: ClusterInstance.provisioned('writer', {
+          writer: ClusterInstance.provisioned('writer'),
+          enablePerformanceInsights: false,
+          performanceInsightRetention: PerformanceInsightRetention.DEFAULT,
+        });
+      }).toThrow(/`enablePerformanceInsights` disabled, but `performanceInsightRetention` or `performanceInsightEncryptionKey` was set/);
+    });
+
+    test('warn if performance insights is enabled at cluster level but disabled on writer instance', () => {
```

Comment:
We could remove this as redundant with `warn if performance insights is enabled at cluster level but disabled on writer and reader instances`

Reply:
done: https://github.com/aws/aws-cdk/pull/31385/commits/a167d1f12169f219eca03b811b50e70b7563566a